### PR TITLE
fix: ensure all rules are recommended by default

### PR DESCRIPTION
**Pull request by nulltoken**
As of Spectral 5, all rules are recommended by default now. However, no normalization is applied to the `recommended` property (as it's done for `severity`, for instance). This PR fixes this.

Also slightly enhance the code coverage to ensure the `recommended` setting of a rule eventually leads to the rule being run or not.

**Checklist**

- [x] Tests added / updated
- [ ] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/rulesets/mergers/rules.ts [hunk 1]
```diff
@@ -117,6 +117,10 @@ function processRule(rules: FileRuleCollection, name: string, rule: FileRule | F
 }
 
 function normalizeRule(rule: Rule, severity: DiagnosticSeverity | HumanReadableDiagnosticSeverity | undefined) {
+  if (rule.recommended === void 0) {
+    rule.recommended = true;
```
Comment (P0lip) [inline src/rulesets/mergers/rules.ts:121]: This is not needed

**Comment by nulltoken** [comment]
> Hm, I might be missing something, but what does this PR fix actually?
> All rules are recommended by default unless you set recommended to false.

@P0lip The first commit that triggered that PR was to actually drop the unneeded `recommended: true` settings in the oas ruleset.

And... tests started to fail.

So I took a look at the normalization process and realized that although we always make `severity` explicit (even when `Warning` is the default), we don't do the same for `recommended`.

This PR attempts at aligning this so that rulesets can be designed with implicit settings and rules, when materialized, always expose explicit settings.

Thoughts?

**Comment by nulltoken** [comment]
> > Hm, I might be missing something, but what does this PR fix actually?
> > All rules are recommended by default unless you set recommended to false.
> 
> @P0lip The first commit that triggered that PR was to actually drop the unneeded `recommended: true` settings in the oas ruleset.
> 
> And... tests started to fail.
> 
> So I took a look at the normalization process and realized that although we always make `severity` explicit (even when `Warning` is the default), we don't do the same for `recommended`.
> 
> This PR attempts at aligning this so that rulesets can be designed with implicit settings and rules, when materialized, always expose explicit settings.
> 
> Thoughts?

@P0lip If you don't think making `recommended` value explicit is worth it, I could also tune down that PR to only focus on dropping the `recommended: true` settings from the oas ruleset and fixing the tests.

**Comment by P0lip** [comment]
> @P0lip The first commit that triggered that PR was to actually drop the unneeded recommended: true settings in the oas ruleset.

Ah, so I see, but in any case I don't think we should drop them - they make everything more readable IMHO. Otherwise a user need to go to docs to see how recommended works.

> And... tests started to fail.

They might have started to fail because we don't normalize the rules, but we still should treat them in the same way, meaning they are not run or run depending on the ruleset you use.

> @P0lip If you don't think making recommended value explicit is worth it, I could also tune down that PR to only focus on dropping the recommended: true settings from the oas ruleset and fixing the tests.

Yeah, I think I'd leave it. IMHO it's just more readable, but it might be just me.


That being said, I'm still open for having a test that actually verifies whether we run a given rule or not. We might have such, but maybe not, so if you could verify that and write one in case we don't, I'd be very grateful

**Comment by nulltoken** [comment]
@P0lip Thanks for the feedback!

So, just to make sure I got your right:
- ok to explicitly normalize the `recommended` setting to `true` while reading the ruleset
- keep the `recommended: true` settings in the oas ruleset
- ensure `recommended` rules are run and not `!recommended` aren't (either find a test that does so or write one)

Would that fit you?

**Comment by P0lip** [comment]
@nulltoken yeah, that works!

**Comment by nulltoken** [comment]
@P0lip I believe I've dealt with all your comments, Sir!

**Pull request merged**